### ISSUE TITLE
`denote-file-is-note-p`, but `t` if file exists *or* buffer visiting file exists

### ISSUE THREAD

**Issue by mentalisttraceur** [protesilaos/denote#277]
TL;DR: I propose adding:

```lisp
(defun denote-file-or-visiting-buffer-is-note-p (file)
    "Return non-nil if FILE is an actual Denote note
(as determined by `denote-file-is-note-p'), or if
there is a buffer visiting FILE and FILE would be
an actual Denote note once that buffer is saved."
    (and (not (file-directory-p file))
         (or (file-regular-p file)
             (find-buffer-visiting file))
         (denote-filename-is-note-p file)))
```

At least in some use-cases, when checking if a file is a note, notes which are merely not-yet-saved should count as notes.

After all, that *is* a note file, conceptually. I know it's a note file, Emacs knows it's a note file, only the file system doesn't know it's a note file.

For example, I have a function which grabs the title either from the front matter or from the file name (but *without* deslugifying the title, because that makes more sense to me), and I noticed some code was wrongly using the filename title instead of the front matter, if I just created a note but hadn't saved it yet. So now I'm using the above function instead of either is-filename-p or is-file-p there, to hit that sweet spot of intuitive/expected/reasonable (to me anyway) behavior.

**Comment by mentalisttraceur** [comment]
By the way, I don't think you need the `(not (file-directory-p file))` line - I left it in for consistency and because maybe you want to make it more explicit, but to my knowledge file-regular-p and file-directory-p are always mutually exclusive.

**Comment by jeanphilippegg** [comment]
Though this function could be useful, I don't think that we need it in Denote's code. In many places, we now use `denote-filename-is-note-p` instead of `denote-file-is-note-p` because we don't need to check that the file actually exists. I don't think we need to test that the file is visited in a buffer.

Maybe your original intention was to handle files that are still not saved? The code should now handle this better.

Would you consider this issue done (in a way)?

**Comment by mentalisttraceur** [comment]
@jeanphilippegg yeah I'm happy to close this.

The use-case I had for this is currently handled just fine by existing Denote code and the `denote-file-note-type` function suggested in #284 :

```lisp
    (defun denoted-title-get (path)
        (let ((note-type (denote-file-note-type path)))
            (if note-type
                (denote-retrieve-title-value path note-type)
                (let ((name (file-name-nondirectory path)))
                    (when (string-prefix-p "." name)
                        (setq path (substring name 1))))
                (if (denote-file-has-identifier-p path)
                    (denote-retrieve-filename-title path)
                    (denote-retrieve-filename-title
                        (denoted--add-nil-id path))))))
```
